_posts/2020-10-27-mutiny-back-pressure.adoc

```diff
@@ -0,0 +1,314 @@
+You can use `onSubscribe().invoke()` and `onRequest().invoke()` to check what's going on:
+
+[source, java, indent=0]
+----
+Multi.createFrom().range(0, 10)
+    .onSubscribe().invoke(sub -> System.out.println("Received subscription: " + sub))
+    .onRequest().invoke(req -> System.out.println("Got a request: " + req))
+    .transform().byFilteringItemsWith(i -> i % 2 == 0)
+    .onItem().transform(i -> i * 100)
+    .subscribe().with(
+            i -> System.out.println("i: " + i)
+);
+---- 
+
+You can also going a step further and not only observing but also controlling the flow yourself:
+
+[source, java, indent=0]
+----
+Multi.createFrom().range(0, 10)
+    .onSubscribe().invoke(sub -> System.out.println("Received subscription: " + sub))
+    .onRequest().invoke(req -> System.out.println("Got a request: " + req))
+    .onItem().transform(i -> i * 100)
+    .subscribe().with(
+            subscription -> {
+                // Got the subscription
+                upstream.set(subscription);
+                subscription.request(1);
+            },
+            i -> {
+                System.out.println("i: " + i);
+                upstream.get().request(1);
+            },
+            f -> System.out.println("Failed with " + f),
+            () -> System.out.println("Completed")
+);
+----
+
+And, if you are a bit insane, you can even implement a `Subscriber` directly:
+
+[source, java, indent=0]
+----
+Multi.createFrom().range(0, 10)
+    .onSubscribe().invoke(sub -> System.out.println("Received subscription: " + sub))
+    .onRequest().invoke(req -> System.out.println("Got a request: " + req))
+    .onItem().transform(i -> i * 100)
+    .subscribe().withSubscriber(new Subscriber<Integer>() {
+
+        private Subscription subscription;
+
+        @Override
+        public void onSubscribe(Subscription s) {
+            this.subscription = s;
+            s.request(1);
+        }
+
+        @Override
+        public void onNext(Integer item) {
+            System.out.println("Got item " + item);
+            subscription.request(1);
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            // ...
+        }
+
+        @Override
+        public void onComplete() {
+            // ...
+        }
+    }
+);
+----        
+
+
+### Summary
+
+This post described the different approaches offered by Mutiny to handle back-pressure. 
+The Reactive Streams protocol works well when you can control the pace of the upstream.
+But it's not always the case. 
+Streams representing physical entities are rarely controllable. 
+Imagine a stream emitting user's keystrokes.
+You cannot ask the users to be slow down. 
+That would be terrible user experience. 
+As we have seen above, time is also not something we can slow down, unfortunately...
+In this case, the `onOverflow()` group lets you decide the mitigation, such as using buffers or dropping items. 
+
+It's imperative to avoid overwhelming downstream subscribers. 
```

Comment:
I'd maybe change "imperative" to "critical", or something similar, to prevent any possible confusion around imperative vs reactive code.

Reply:
good point!